include/ddc/kernels/splines/matrix_maker.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <memory>
+
+#include "Kokkos_Core_fwd.hpp"
+#include "matrix_sparse.hpp"
+
+
+namespace ddc::detail {
+class MatrixMaker
+{
+public:
+    template <typename ExecSpace>
+    static std::unique_ptr<Matrix> make_new_sparse(int const n)
+    {
+        return std::make_unique<Matrix_Sparse<ExecSpace>>(n);
+    }
+};
+} // namespace ddc::detail
```

Comment:
I would simplify like this

```suggestion
namespace ddc::detail {

template <typename ExecSpace>
std::unique_ptr<Matrix> make_new_sparse(int const n)
{
    return std::make_unique<Matrix_Sparse<ExecSpace>>(n);
}

} // namespace ddc::detail
```

Reply:
I can't do this without making the different instances of sll diverge (in Gysela and in splines branch). Let's keep this change for later.

Reply:
I think they already diverge don't they ?

Reply:
Not really, for example non-periodic tests runs, PolarSplineBuilder and PolarSplineEvaluator still compile in my working branch "splines" etc. I really want to avoid breaking changes until everything have been ported to GPU. Moreover I will have to integrate the changes from Pauline, which are in Gysela/sll and it will be far easier to do so before beginning with the GPU port of polar splines.